src/Integration.Vsix.UnitTests/CFamily/ProtocolTest.cs

```diff
@@ -71,6 +71,28 @@ public void Read_Empty_Response()
             response.Messages.Length.Should().Be(0);
         }
 
+        [TestMethod]
+        public void Read_RequestHasFileName_ReturnedMessageHasNoFileName_MessageIgnored()
+        {
+            const string returnedIssueFileName = "";
+            const string testedFileName = "myfile.cpp";
+
+            var response = CallProtocolRead(MockResponse(returnedIssueFileName), testedFileName);
+
+            response.Messages.Length.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void Read_RequestHasNoFileName_ReturnedMessageHasNoFileName_MessageReturned()
+        {
+            const string returnedIssueFileName = "";
+            const string testedFileName = "";
+
+            var response = CallProtocolRead(MockResponse(returnedIssueFileName), testedFileName);
+
+            response.Messages.Length.Should().Be(1);
+        }
+
```

Comment:
Are there already tests for the other two cases? i.e. both have same file name, both have different file names

Reply:
Yep. Same name is "Response_Filtering_WithFiltering", and different name is "Response_Filtering_NoMatches"